### PR TITLE
Push to Github Registry 

### PR DESCRIPTION
**Pull request by EnriqueL8**
## Proposed changes

Push images to Github Registry 

Builds on top of #377 

Example run https://github.com/EnriqueL8/paladin/actions/runs/11685400404/job/32538740521

<hr>

## Please make sure to follow these points 

- [x] I have read the contributing guidelines.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

<hr>

## Screenshots

Tested on fork, you can see two Arch images published

image

### CODE REVIEW

File: .github/workflows/build-image-job.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+      - name: Docker registry login
+        if: ${{ inputs.push }}
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ inputs.registry }}
+          username: ${{ secrets.username }}
+          password: ${{ secrets.password }}
+
+      # https://github.com/orgs/community/discussions/27086
```
Comment (dwertent) [inline .github/workflows/build-image-job.yaml:90]: This workaround is only for ghcr, so it should look more like:
```
      # https://github.com/orgs/community/discussions/27086
      - name: Set lower case image name
        if ${{ inputs.registry }} == "ghcr.io"
        run: |
          echo "IMAGE=${IMAGE,,}" >>${GITHUB_ENV}
        env:
          IMAGE: ${{ inputs.registry }}/${{ inputs.image }}
```

**Review by dwertent — CHANGES_REQUESTED** [review]

File: .github/workflows/build-image-job.yaml (hunk 1 above)
Comment (EnriqueL8) [reply .github/workflows/build-image-job.yaml:90]: Also for Docker it's needed

Comment (EnriqueL8) [reply .github/workflows/build-image-job.yaml:90]: https://docs.docker.com/docker-hub/repos/create/#:~:text=The%20repository%20name%20needs%20to,)%2C%20and%20underscores%20(%20_%20)

**Comment by dwertent** [comment]
I opened #383 instead

**Pull request closed without merging**